test/e2e/far_e2e_test.go

```diff
@@ -240,24 +250,36 @@ func buildNodeParameters(clusterPlatformType configv1.PlatformType) (map[v1alpha
 	return testNodeParam, nil
 }
 
-// getAvailableNodes a list of available worker nodes in the cluster
-func getAvailableWorkerNodes() *corev1.NodeList {
-	availableNodes := &corev1.NodeList{}
+// getReadyWorkerNodes returns a list of ready worker nodes in the cluster if any
+func getReadyWorkerNodes() *corev1.NodeList {
+	availableWorkerNodes := &corev1.NodeList{}
 	selector := labels.NewSelector()
-	requirement, _ := labels.NewRequirement(medik8sLabels.WorkerRole, selection.Exists, []string{})
+	requirement, err := labels.NewRequirement(medik8sLabels.WorkerRole, selection.Exists, []string{})
+	Expect(err).To(BeNil())
 	selector = selector.Add(*requirement)
-	Expect(k8sClient.List(context.Background(), availableNodes, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
-	if len(availableNodes.Items) < 1 {
-		Fail("No worker nodes found in the cluster")
+	Expect(k8sClient.List(context.Background(), availableWorkerNodes, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
+
+	// Filter nodes to only include those in "Ready" state
+	readyWorkerNodes := &corev1.NodeList{}
+	for _, node := range availableWorkerNodes.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+				readyWorkerNodes.Items = append(readyWorkerNodes.Items, node)
+				break // "Ready" was found
+			}
+		}
+	}
+	if len(readyWorkerNodes.Items) < 1 {
+		Skip("There isn't an available (and ready) worker node in the cluster")
 	}
-	return availableNodes
+	return readyWorkerNodes
```

Comment:
nit: why returning a list when we only add max one to it?

Reply:
>when we only add max one to it?

Can you elaborate? The function returns the list of Ready worker nodes. Assuming there are 3 worker nodes in the cluster, then the returned list can be at most 3 nodes.

Reply:
ah sorry, my fault, the `break` in the loop refers to the conditions only